src/Chainweb/Rosetta/RestAPI/Server.hs

```diff
@@ -151,9 +181,142 @@ mempoolTransactionH ms mtr = runExceptT work >>= either throwRosetta pure
     work :: ExceptT RosettaFailure Handler MempoolTransactionResp
     work = do
         SubNetworkId n _ <- msni ?? RosettaChainUnspecified
-        mp <- (readMaybe (T.unpack n) >>= flip lookup ms) ?? RosettaInvalidChain n
+        mp <- (readChainIdText v n >>= flip lookup ms) ?? RosettaInvalidChain n
         lrs <- liftIO . mempoolLookup mp $ V.singleton th
         (lrs V.!? 0 >>= f) ?? RosettaMempoolBadTx
 
 --------------------------------------------------------------------------------
 -- Network Handlers
+
+networkListH :: ChainwebVersion -> MetadataReq -> Handler NetworkListResp
+networkListH v _ = pure $ NetworkListResp networkIds
+  where
+    -- Unique Rosetta network ids for each of the chainweb version's chain ids
+    networkIds = map f (HS.toList (chainIds v))
+    f :: ChainId -> NetworkId
+    f cid =  NetworkId
+      { _networkId_blockchain = blockchainName
+      , _networkId_network = (chainwebVersionToText v)
+      , _networkId_subNetworkId = Just (SubNetworkId (chainIdToText cid) Nothing)
+      }
+
+networkOptionsH :: ChainwebVersion -> NetworkReq -> Handler NetworkOptionsResp
+networkOptionsH v (NetworkReq nid _) = do
+  _ <- handle (enforceValidNetworkId v nid)
+  pure $ NetworkOptionsResp version allow
+  where
+    version = RosettaNodeVersion
+      { _version_rosettaVersion = "1.3.1"   -- TODO: Make this a variable in /rosetta repo
+      , _version_nodeVersion = chainwebNodeVersionHeaderValue :: T.Text
```

Comment:
@larskuhtz do you suggest using the chainweb node version (i.e. 1.8) here? I've included the api version and the chainweb version in the meta data section as well. 
This field is supposed to be:
>The node_version is the canonical version of the node runtime. This can help clients manage deployments.